versioned_docs/version-V3/guides/governance/license-modifications.md

```diff
@@ -0,0 +1,171 @@
+
+Note: [`setContentHash`](https://docs.ens.domains/contract-api-reference/publicresolver#set-content-hash) may also be used to associate text with a subdomain, but `setText` is presented above for simplicity.
+
+These contract function calls should then be encoded into a governance proposal, and approved by Uniswap Governance.
+
+</details>
+
+## Proposals
+
+Proposals are submitted to Uniswap Governance via the [Governance Bravo Delegate contract](https://etherscan.io/address/0x53a328f4086d7c0f1fa19e594c9b842125263026#code). NPM packages for consuming the governance contract ABIs, and details on previous versions, are available [here](https://docs.uniswap.org/protocol/concepts/governance/overview).
+
+<details>
+    <summary> Governor Bravo #propose Parameters </summary>
+
+Deployment reference available [here.](https://etherscan.io/address/0x53a328f4086d7c0f1fa19e594c9b842125263026#code)
+
+```solidity
+/**
+    * @notice Function used to propose a new proposal. Sender must have delegates above the proposal threshold
+    * @param targets Target addresses for proposal calls
+    * @param values Eth values for proposal calls
+    * @param signatures Function signatures for proposal calls
+    * @param calldatas Calldatas for proposal calls
+    * @param description String description of the proposal
+    * @return Proposal id of new proposal
+    */
+function propose(
+    address[] memory targets,
+    uint[] memory values,
+    string[] memory signatures,
+    bytes[] memory calldatas,
+    string memory description
+) public returns (uint)
+
+```
+
+</details>
+
+## Populating Proposal Calldata
+
+Below is an example of using a scripting environment to generate a proposal. This is for educational purposes only - that example assumes access to a private key with a sufficient amount of delegated UNI to submit a proposal, which may be an insecure practice. There are several ways to generate a proposal transaction and submit it to Ethereum; this example should only be used for reference and not in production.
+
+<details>
+<summary> Populating `Propose` Calldata </summary>
+
+```typescript
+import { ethers } from "hardhat";
+import { waffle } from "hardhat";
+import { Contract } from "ethers";
+import {
+  GOVERNOR_BRAVO_ABI,
+  ENS_REGISTRY_ABI,
+  ENS_PUBLIC_RESOLVER_ABI,
+} from "./utils";
+import { namehash } from "@ethersproject/hash";
+import { keccak256 } from "@ethersproject/keccak256";
+import { utils } from "ethers";
+import { Interface } from "@ethersproject/abi";
+import "hardhat";
+
+const { provider } = waffle;
+
+// get the governor bravo contract
+const governorBravoAddress = "0x408ED6354d4973f66138C91495F2f2FCbd8724C3";
+const governorBravo = new Contract(
+  governorBravoAddress,
+  GOVERNOR_BRAVO_ABI,
+  provider
+);
+
+const NODE_TOP_LEVEL: string = namehash("uniswap.eth");
```

Comment:
all constants should be declared at the top - i recognize there's a lot of them and it's nice to have them declared just above where they're used but i think it's better to be consistent